renderservice/render/daemon/FileManager.py

```diff
@@ -0,0 +1,95 @@
+"""Handles access to external filesystems from the render service."""
+import os
+from cachetools import LRUCache, cachedmethod
+from io import BytesIO
+
+CACHE_SIZE = int(os.getenv("DAEMON_CACHE_SIZE", 300 * 1024 * 1024))
+
+
+class FileManager(object):
+    """Handles access to the filesystem for loading and reading files.
+
+    Caching is important to this class as it speeds up slow reads when
+    the directory is a mounted bucket.
+
+    Since files with the same filepath could belong to different
+    directories only the first found is used, based on the order
```

Comment:
The render system has two locations for static files. A local copy for do don't distribute resources and things for speed, and the cloud bucket system for all other static files. Since the bucket is mounted as another folder in the file-system this class enables the use of both these locations as a single entity.